major and minor are defined by sys/mkdev.h on Solaris

--- shlibs/blkid/src/topology/md.c
+++ shlibs/blkid/src/topology/md.c
@@ -9,6 +9,10 @@
  * GNU Lesser General Public License.
  *
  */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <errno.h>
 #include <fcntl.h>
 #include <stdint.h>
@@ -16,6 +20,9 @@
 #include <sys/ioctl.h>
 #include <sys/stat.h>
 #include <sys/types.h>
+#ifdef HAVE_SYS_MKDEV_H
+#include <sys/mkdev.h>
+#endif
 #include <unistd.h>
 
 #include "topology.h"
--- shlibs/blkid/src/topology/evms.c
+++ shlibs/blkid/src/topology/evms.c
@@ -9,6 +9,10 @@
  * GNU Lesser General Public License.
  *
  */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <errno.h>
 #include <fcntl.h>
 #include <stdint.h>
@@ -16,8 +20,14 @@
 #include <stdlib.h>
 #include <string.h>
 #include <sys/ioctl.h>
+#ifdef HAVE_SYS_IOCCOM_H
+#include <sys/ioccom.h>
+#endif
 #include <sys/stat.h>
 #include <sys/types.h>
+#ifdef HAVE_SYS_MKDEV_H
+#include <sys/mkdev.h>
+#endif
 #include <unistd.h>
 
 #include "topology.h"
--- shlibs/blkid/src/topology/lvm.c
+++ shlibs/blkid/src/topology/lvm.c
@@ -9,6 +9,10 @@
  * GNU Lesser General Public License.
  *
  */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <errno.h>
 #include <fcntl.h>
 #include <stdint.h>
@@ -17,6 +21,9 @@
 #include <string.h>
 #include <sys/stat.h>
 #include <sys/types.h>
+#ifdef HAVE_SYS_MKDEV_H
+#include <sys/mkdev.h>
+#endif
 #include <unistd.h>
 
 #include "topology.h"
--- shlibs/blkid/src/topology/dm.c
+++ shlibs/blkid/src/topology/dm.c
@@ -9,6 +9,10 @@
  * GNU Lesser General Public License.
  *
  */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <errno.h>
 #include <fcntl.h>
 #include <stdint.h>
@@ -17,6 +21,9 @@
 #include <string.h>
 #include <sys/stat.h>
 #include <sys/types.h>
+#ifdef HAVE_SYS_MKDEV_H
+#include <sys/mkdev.h>
+#endif
 #include <unistd.h>
 
 #include "topology.h"
--- shlibs/blkid/src/partitions/partitions.c
+++ shlibs/blkid/src/partitions/partitions.c
@@ -7,6 +7,10 @@
  * GNU Lesser General Public License.
  *
  */
+#if HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
@@ -14,6 +18,9 @@
 #include <fcntl.h>
 #include <ctype.h>
 #include <sys/types.h>
+#if HAVE_SYS_MKDEV_H
+#include <sys/mkdev.h>
+#endif
 #include <sys/stat.h>
 #include <errno.h>
 #include <stdint.h>
